hash.c

```diff
@@ -4420,33 +4482,76 @@ rb_hash_update_by(VALUE hash1, VALUE hash2, rb_hash_update_func *func)
 
 /*
  *  call-seq:
- *     hsh.merge(other_hash1, other_hash2, ...)           -> new_hash
- *     hsh.merge(other_hash1, other_hash2, ...) {|key, oldval, newval| block}
- *                                                        -> new_hash
- *
- *  Returns a new hash that combines the contents of the receiver and
- *  the contents of the given hashes.
- *
- *  If no block is given, entries with duplicate keys are overwritten
- *  with the values from each +other_hash+ successively,
- *  otherwise the value for each duplicate key is determined by
- *  calling the block with the key, its value in the receiver and
- *  its value in each +other_hash+.
- *
- *  When called without any argument, returns a copy of the receiver.
- *
- *     h1 = { "a" => 100, "b" => 200 }
- *     h2 = { "b" => 246, "c" => 300 }
- *     h3 = { "b" => 357, "d" => 400 }
- *     h1.merge          #=> {"a"=>100, "b"=>200}
- *     h1.merge(h2)      #=> {"a"=>100, "b"=>246, "c"=>300}
- *     h1.merge(h2, h3)  #=> {"a"=>100, "b"=>357, "c"=>300, "d"=>400}
- *     h1.merge(h2) {|key, oldval, newval| newval - oldval}
- *                       #=> {"a"=>100, "b"=>46,  "c"=>300}
- *     h1.merge(h2, h3) {|key, oldval, newval| newval - oldval}
- *                       #=> {"a"=>100, "b"=>311, "c"=>300, "d"=>400}
- *     h1                #=> {"a"=>100, "b"=>200}
+ *    hash.merge -> new_copy_of_hash
+ *    hash.merge(*other_hashes) -> new_hash
+ *    hash.merge(*other_hashes) { |key, old_value, new_value| ... } -> new_hash
+ *
+ *  Each argument in +other_hashes+ must be
+ *  a {Hash-convertible object}[doc/implicit_conversion_rdoc.html#label-Hash-Convertible+Objects].
+ *
+ *  ---
+ *
+ *  With arguments and no block:
+ *  * Returns a new \Hash object that is the merge of +self+ and each given hash.
```

Comment:
"is the merge of +self+ and each given hash" doesn't seem quite right. Can we rewrite it to be clearer?

Reply:
Re-written.